home/tests/test_display_question_feature.py

```diff
@@ -114,3 +114,15 @@ def test_sorting_answers(self, client, filterType, expected):
             response = client.get(url)
             answer = response.context['answers'].first()
             assert str(answer.content) == expected
+
+        @pytest.mark.django_db
+        def test_invalid_question_url(self, client):
+            url = reverse('question-detail', args=[999])
+            response = client.get(url)
+            assert response.status_code == 404
```

Comment:
he doesn't get a message.

Reply:
prior to this PR, when an invalid URL was entered, you would get a server-error response (status 500) 
and I think getting 404 is better and it usually means to the user that he entered a URL that doest not exist

Reply:
I agree with you that it would be better for the status to be 404 than 500 as it is now.
According to the updated version I am currently seeing, the message "Question matching query does not exist" is printed as soon as you reach a question page that does not exist (such as 999).
If this message still appears after your update, I think you should add a test that checks that this is the message that appears on the error page (And not something else like "The answer does not exist, log in details are incorrect, etc ...").
You can do this by checking the content of the page as soon as you reach this page.

Reply:
thanks for the review, but there is no message, it's a plain 404 page so there's nothing to test

Reply:
I believe the test coverage here is good.